### PR TITLE
IDEMPIERE-5275 - Tabular Report Re-Query button/close paramater window

### PR DESCRIPTION
**Pull request by PeterTakacs300**
JIRA ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5275

### CODE REVIEW

**Comment by norbertbede** [comment]
notes:  Design, Architect: @igorpojzl  Coding/Commit: @PeterTakacs300

**Comment by norbertbede** [comment]
image

an example. tested many ways for community compactibility (the screencast was done on private version, the code is for latest v10)

**Review by hengsin — CHANGES_REQUESTED** [review]
2 issue with the rerun dialog
- the initial size of the dialog is not optimal, some options need horizontal scroll to access.
- the cancel button is not working correctly.

otherwise, seems to work just fine.

**Comment by norbertbede** [comment]
changes

1. sysconfig: ZK_REPORT_TABLE_OPEN_IN_NEW_TAB - default No - still possible to run re-run from toolbar - but doesn't confuse users with breaking change
2. cancel button fixed
3. scroll resolved - increased with. 

norbert

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Hi @PeterTakacs300 

Can you please include the following patch:
[1306_PeerReview.txt](https://github.com/idempiere/idempiere/files/8677136/1306_PeerReview.txt)


It adds the following changes:
- change the SysConfig scope to Tenant
- change the text of the message to "Run it again with new parameters"
- add the definition of the new button in Toolbar (this way it can be restricted per role)
- change timestamp in lang-addon.xml (because a css was changed)
- reorder font-icons.css
- reorder MSysConfig and generate serialVersionUID
- remove a warning on ReportCtl

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/ZkReportViewer.java
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/ZkReportViewer.java:1252]: if this is means to run hideBusyMask() if event is on_window_close, it is more readable to use a positive condition block instead, i.e if (DialogEvents.ON_WINDOW_CLOSE.equals(event.getName())) hideBusyMask();

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/ZkReportViewer.java
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/ZkReportViewer.java:1502]: can AD_Process_ID be 0 here ? if yes, then the method should stop immediately (i.e return;) if AD_Process_ID is 0

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/ZkReportViewer.java
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/ZkReportViewer.java:1505]: use ZkUpdateUtil.setWindowWidthX(...) instead.

File: org.adempiere.base/src/org/compiere/process/ProcessInfo.java
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/process/ProcessInfo.java:157]: I think it is more readable to use m_IsReplaceTabContent instead

File: org.adempiere.base/src/org/compiere/process/ProcessInfo.java
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/process/ProcessInfo.java:952]: please always add javadoc for public methods. also, as per my comments above, I think it is more readable to use isReplaceTabContent and setReplaceTabContent instead.

File: org.adempiere.base/src/org/compiere/print/ReportEngine.java
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/print/ReportEngine.java:2402]: add javadoc for public methods and I think it is more readable to use IsReplaceTabContent instead of isOverride.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessModalDialog.java
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessModalDialog.java:77]: it seems this is always call with autoStart = false and loadSavedParams = true, if that's the case, simpler and more readable to just remove the 2 process parameter and use the false and true directly.

**Review by hengsin — CHANGES_REQUESTED** [review]

**Comment by PeterTakacs300** [comment]
I think I've applied all the requested changes, thank you for the patch and the detailed review @CarlosRuiz-globalqss and @hengsin

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessModalDialog.java
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessModalDialog.java:76]: wrong javadoc

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Window.java
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Window.java:62]: Comment should be changed to something like "Replace Current Tab Content"

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/part/WindowContainer.java
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/part/WindowContainer.java:653]: what is this for ? Since refTab must not be null here, probably should just throw IllegalArgumentException here.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/part/WindowContainer.java
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/part/WindowContainer.java:666]: the change make for this block is wrong. The previous if (oldWindow instanceof Window) condition is correct, what I mean is you should have the else part too, i.e when oldWindow is not instanceof Window, you should just detach oldWindow (probably sounds more readable if that's rename to firstChild instead) and attach comp to refpanel.

**Review by hengsin — CHANGES_REQUESTED** [review]

**Comment by hengsin** [comment]
@PeterTakacs300 , please resolve the conflict.

**Comment by norbertbede** [comment]
@hengsin, @PeterTakacs300 is out of office will be after 14h CET.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java
Comment (CarlosRuiz-globalqss) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java:316]: WARNING: this is reverting the change from:
https://github.com/idempiere/idempiere/commit/895923f7f229e3c3195c691afdba3c12e6fe61ca#diff-45852062765555bb780d1af8808a6582852fa6a3102af104344b02c5b2bad61c

Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java:316]: this merge with master is wrong (i.e it should be using the (function(){...})()).

**Review by hengsin — CHANGES_REQUESTED** [review]

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java
Comment (PeterTakacs300) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java:316]: yes thank you, fixing in a moment..

**Review by CarlosRuiz-globalqss — APPROVED** [review]
from my side it works fine

let's wait for @hengsin approval too

**Review by hengsin — APPROVED** [review]
looks good and tested by Carlos.

**Pull request merged**
